app/models/issue.rb

```diff
@@ -45,4 +45,16 @@ def lt_ln_present?(obj)
   def status_name
     STATUSES[status]
   end
+
+  def published?
+    (status == 'open') || (status == 'closed')
+  end
+
+  def can_read_by?(user)
+    if user.present?
+      user.admin? || user.moderator? || (self.user == user) || published?
+    else
+      false
+    end
+  end
```

Comment:
This method looks rly bad. Let's think how we can improve it.
1st we can apply NullObject pattern and add to application_controller:
```ruby
def current_user
  @current_user ||= super || Guest.new
end
```

This will allow us to get rid off all ```user.nil?``` checks and than ```def can_read_by?``` method will look like:

```ruby
def can_read_by? user
  published? || user.admin? || user.moderator? || (self.user == user)
end
```